libraries/lib-music-information-retrieval/MirAudioReader.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <cstddef>
+
+namespace MIR
+{
+class MirAudioReader
+{
+public:
+   virtual int GetSampleRate() const = 0;
+   virtual long long GetNumSamples() const = 0;
+   virtual void
+   ReadFloats(float* buffer, long long where, size_t numFrames) const = 0;
+   virtual ~MirAudioReader() = default;
```

Comment:
I would make a small companion `.cpp` file, just to define the destructor non-inline, so that the linker doesn't need to de-duplicate the virtual function table but will allocate it just in that .cpp.

Not very important for a header file that is not widely used, but I make a habit of this whenever I make a new abstract base class.